### PR TITLE
Adding in the `compose2` combinator

### PR DESCRIPTION
**Pull request by dalefrancis88**
per the discussion on gitter and https://github.com/evilsoft/crocks/issues/428 this is to add in the `compose2` combinator

This will allow for some nice composition

### CODE REVIEW

File: docs/src/pages/docs/functions/combinators.md [hunk 1]
```diff
@@ -64,6 +64,119 @@ safeYell(42)
...
+
+```haskell
+compose2 :: (c -> d -> e) -> (a -> c) -> (b -> d) -> a -> b -> e
+```
+
+`compose2` allows for composition between a `binary` function and
+two `unary` functions. `compose2` takes a `binary` function followed by
+two `unary` functions and returns a `binary` function that maps the first
+argument with the first `unary` and the second with the second, passing
+the results to the given `binary` and returning the result.
```
Comment (dalefrancis88) [inline docs/src/pages/docs/functions/combinators.md]: Really not married to this description, it feels a little flat, lmk if anyone can make it better

File: docs/src/pages/docs/functions/combinators.md [hunk 2]
```diff
@@ -64,6 +64,119 @@ safeYell(42)
...
+//=> Nothing
+
+// Item :: { id: Integer }
+// Items :: Array Item
+const items =
+  [ { id: 2 }, { id: 1 } ]
+
+// pluck :: String -> Array Object -> Maybe a
+const pluck =
+  compose2(applyTo, prop, flip(map))
```
Comment (dalefrancis88) [inline docs/src/pages/docs/functions/combinators.md]: This is to show usage in relation to https://github.com/evilsoft/crocks/issues/338

File: docs/src/pages/docs/functions/combinators.md (hunk 1 above)
Comment (evilsoft) [reply docs/src/pages/docs/functions/combinators.md]: I think this is good as is, gets to the point and provides the info someone would need to wrap their head around it.

File: docs/src/pages/docs/functions/combinators.md [hunk 3]
```diff
@@ -64,6 +64,119 @@ safeYell(42)
 //=> Left 42
 ```
 
+#### compose2
```
Comment (evilsoft) [inline docs/src/pages/docs/functions/combinators.md]: This should come before `composeB`

File: docs/src/pages/docs/functions/combinators.md [hunk 4]
```diff
@@ -64,6 +64,119 @@ safeYell(42)
...
+import hasProp from 'crocks/predicates/hasProp'
+import isNumber from 'crocks/predicates/isNumber'
+import liftA2 from 'crocks/helpers/liftA2'
+import map from 'crocks/pointfree/map'
+import prop from 'crocks/Maybe/prop'
+import safe from 'crocks/Maybe/safe'
+import safeLift from 'crocks/Maybe/safeLift'
+
+// isNonZero :: Number -> Boolean
+const isNonZero = x => x !== 0
```
Comment (evilsoft) [inline docs/src/pages/docs/functions/combinators.md]: These should all be multi-line
```js
// isNonZero :: Number -> Boolean
const isNonZero = x =>
  x !== 0

// isValidDivisor :: Number -> Boolean
const isValidDivisor =
  and(isNumber, isNonZero)

// divideBy :: Number -> Number -> Number
const divideBy = x => y =>
  y / x

// safeDivide :: Number -> Number -> Maybe Number
const safeDivide = compose2(
  liftA2(divideBy),
  safe(isValidDivisor),
  safe(isNumber)
)
```

File: docs/src/pages/docs/functions/combinators.md [hunk 5]
```diff
@@ -64,6 +64,119 @@ safeYell(42)
...
+const createSummary = compose2(
+  liftA2(summarize),
+  prop('name'),
+  getLength
+)
+
+createSummary({
+  name: 'Sam Smith'
+}, items)
+// => Just "Sam Smith purchased 2 items"
```
Comment (evilsoft) [inline docs/src/pages/docs/functions/combinators.md]: extra space after comment

File: src/combinators/compose2.spec.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  const fn = bindFunc(compose2)
+  const f = x => y => x * y
+  const g = x => x - 1
+  const h = x => x + 1
+  const x = 22
+  const y = 1
+
+  t.ok(isFunction(compose2), 'is a function')
+
+  const err = /TypeError: compose2: First, second and third arguments must be functions/
```
Comment (evilsoft) [inline src/combinators/compose2.spec.js]: Interesting, including `TypeError`. Thinking we should do this for all cases.
should this include the `^` at the start, JIC.
```js
const err = /^TypeError: compose2: First, second and third arguments must be functions/
```

Comment (dalefrancis88) [reply src/combinators/compose2.spec.js]: It doesn't hurt to add the ^ and i think being as specific as possible without being limiting is the best call

**Pull request merged**